### ISSUE TITLE
core functionality in separate package

### ISSUE THREAD

**Issue by tpapp** [peterkovesi/PerceptualColourMaps.jl#15]
I am wondering if the core functionality (mostly in `src/cmap.jl`) could be factored out into a smaller package with lighter dependencies (especially dropping PyPlot, which is rather heavy and is occasionally tricky to install). Ideally, the package would just export `cmap`, which would return a colormap.

**Comment by peterkovesi** [comment]
Yes, it would make sense to partition the package into two, I had been vaguely thinking about this for a while but have been busy with other things.  As you suggest one could have a lightweight package for providing colour maps, some functions for simple  transformations on colour maps (reversal, resampling, cyclic shift etc) and perhaps some functions for applying colour maps to data (mainly for cyclic and diverging maps).  One would then have a second package for designing colour maps.  This second package would require plotting for providing diagnostic information in the design of the maps.  I have used PyPlot simply out of habit and because for my other work I often need 3D plotting and the plotting of points and lines over images. 

For MATLAB I have an automatically generated standalone function that just provides the colour maps.  <https://www.peterkovesi.com/matlabfns/Colourmaps/colorcet.m> .  The colour maps are provided as 256x3 arrays of values defined in the function body.  However, when one defines all the colour maps the function ends up being 500k which is a bit clumsy and large.  I could have all the colour maps concisely specified in terms of their original design parameters but then some computation would be required to construct each one before use.  Do you have any feelings about this either way?

**Comment by mkborregaard** [comment]
@tpapp See also the discussion here: https://github.com/peterkovesi/PerceptualColourMaps.jl/issues/4
@peterkovesi See also the discussion here: https://discourse.julialang.org/t/ann-fatou-jl-easily-share-julia-fractals/12278/25 and below :-)
